sycl/doc/extensions/experimental/sycl_ext_oneapi_cuda_async_barrier.asciidoc

```diff
@@ -0,0 +1,380 @@
+
+`arrival_token arrive_no_complete(int32_t count)` : Executes arrival operation
+ that counts as `count` arrivals and returns a token that is needed for the 
+ `wait` call corresponding to this arrival. If this is the last arrival that 
+ causes the cycle to complete, behavior is undefined. That means `count` must 
+ be strictly lower than the remaining number of arrivals required to complete 
+ this cycle.
+
+This can be used to signal many arrivals by one function call. However, it 
+should not be used on its own, as it can not be the last arrival in a cycle. 
+So it should either be followed with a call to `arrive` or by a barrier 
+operation, such as (such as `group_barrier()`) after which a different 
+work-item is guaranteed to call `arrive` within the same cycle. 
+
+`arrival_token arrive_and_drop_no_complete(int32_t count)` : Reduces expected 
+arrival count for future cycles by `count`, executes arrival operation that 
+counts as `count` arrivals and returns a token that is needed for the wait 
+call corresponding to this arrival. This must not be the last arrival thet 
+causes the cycle to complete - it would be undefined behavior. That means 
+`count` must be strictly lower than the remaining number of arrivals required 
+to complete this cycle.
+
+This can be used to signal many arrivals by one function call. However, it 
+should not be used on its own, as it can not be the last arrival in a cycle. 
+So it should either be followed with a call to `arrive` or by a barrier 
+operation, such as (such as `group_barrier()`) after which a different 
+work-item is guaranteed to call `arrive` within the same cycle. 
+
+`void arrive_copy_async()` : Schedules arrive operation to be triggered
+asynchronously when all previous asynchronous memory copies initiated by the 
+calling work item complete. Before the arrive operation is triggered, the 
+pending count on the barrier is increased by 1, so after the arrival there is 
+no change to the pending count. Pending count with the increase by this call 
+must not exceed the value returned by `max`. If it does, it causes undefined 
+behavior.
+
+`void arrive_copy_async_no_inc()` : Schedules arrive operation to be triggered 
+asynchronously when all previous asynchronous memory copies initiated by the 
+calling work item complete.
+
+`void wait(arrival_token arrival)` : Executes wait operation, blocking until 
+the predetermined number of work items have called `arrive`.
+
+`bool test_wait(arrival_token arrival)` : Checks whether all the arrivals have 
+already happened for the current cycle, returning `true` if they did and 
+`false` if `wait(arrival)` would block.
+
+`void arrive_and_wait()` : Equivalent to calling `wait(arrive())`.
+
+`static constexpr uint64_t max()` : Returns the maximum value of the expected 
+and pending counts supported by the implementation.
+
+==== Sample Header
+
+[source, c++]
+----
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace cuda {
```

Comment:
If this is an experimental extension, it needs to be defined in the namespace `sycl::ext::oneapi::experimental::cuda`.

Reply:
done

Reply:
fixed